docs/specs/code/COIN_PRICE.md

```diff
@@ -0,0 +1,169 @@
+# Abstract
+
+Describe the ability to see and input ADA amount in another currency.
+
+# Proposal
+
+This feature includes two aspects:
+
+## 1. Input
+When the UI contains an ADA amount input, it should let the user input the number in another currency and covert it to ADA amount at the present conversion rate.
+
+## 2. Output
+When the UI shows an ADA amount, additionally it should be able to show the amount in another currency. If the ADA amount is related to an existing transaction, the user will be able to see the price at two different moments:
+-right now
+-at the moment of the transaction.
+
+The following fiat currencies will be supported:
+
+- USD	US dollar	
+- JPY	Japanese yen
+- EUR   Euro
+- CNY	Chinese yuan renminbi
+- KRW	South Korean won
+
+These are the currencies supported by https://exchangeratesapi.io/. See the Architecture section for details.
+
+The following cryptos will be supported:
+- BTC
+- ETH
```

Comment:
Does this mean ADA/BTC and ADA/ETH will be supported? If so, did you consider Binance API? There is a public endpoint for average ADABTC and ADAETH prices: 

* https://api.binance.com/api/v3/avgPrice?symbol=ADABTC
* https://api.binance.com/api/v3/avgPrice?symbol=ADAETH

According to [this](https://api.binance.com/api/v1/exchangeInfo), there is a limit of 20 weighted requests per second. The weight of the price request is 1, so 20 price requests per second (in theory). This way you won't have to make a double conversion from BTC-USD->USD-ADA and you can distribute the load to avoid surpassing the 100,000 limit of cryptocompare API.

More info: https://github.com/binance-exchange/binance-official-api-docs/blob/master/rest-api.md

Reply:
Instead of doing ADA/BTC directly, everything is converted to USD. It lowers the amount of pairs we need to store from `n^2` to just `n`